gateway/gateway.go

```diff
@@ -266,9 +266,15 @@ func (gw *Gateway) handleMessage(msg config.Message, dest *bridge.Bridge) []*BrM
 		return brMsgIDs
 	}
 
-	// only relay topic change when configured
-	if msg.Event == config.EVENT_TOPIC_CHANGE && !gw.Bridges[dest.Account].GetBool("ShowTopicChange") {
-		return brMsgIDs
+	if msg.Event == config.EVENT_TOPIC_CHANGE {
+		// don't relay topic/purpose change if disabled
+		if !gw.Bridges[dest.Account].GetBool("ShowTopicChange") {
+			return brMsgIDs
+		}
+		// don't relay topic/purpose update if sync is enabled, but msg marked as nosync
```

Comment:
This logic is not necessary. The Slack receiving handler logic added in this PR already takes care of this case.

Reply:
That receive handler drops based on the _existing_ topic/purpose having [nosync] flag set, as that channel shouldn't be overridden by an incoming topic update msg. This logic, on the other hand, keeps a topic update msg in one channel from propagating to others, as it should just be dropped.

I was torn, as this separation is admittedly confusing but seems the proper way (?) because sync can technically be enabled/disabled per-bridge. Open to other ideas!

Reply:
Have indeed added a comment in the receiving handler. We should be handling the `nosync` there. The idea is to kill the message propagation in the bridge as soon as we can. Such specific logic should not be part of the gateway.